rust_snuba/src/processors/generic_metrics.rs

```diff
@@ -91,30 +92,55 @@ enum MetricValue {
 
 #[derive(Debug, Deserialize)]
 #[serde(tag = "format", rename_all = "lowercase")]
-enum EncodedSeries<T> {
+enum EncodedSeries<T: Decodable> {
     Array { data: Vec<T> },
+    Base64 { data: String },
 }
 
-impl<T> EncodedSeries<T> {
+impl<T: Decodable> EncodedSeries<T> {
     fn into_vec(self) -> Vec<T> {
         match self {
             EncodedSeries::Array { data } => data,
+            EncodedSeries::Base64 { data } => {
+                let v = BASE64.decode(data.as_bytes()).ok().unwrap();
+                v.chunks_exact(8)
+                    .map(TryInto::try_into)
+                    .map(Result::unwrap)
+                    .map(T::decode_bytes)
+                    .collect()
+            }
         }
     }
 }
 
+trait Decodable {
+    fn decode_bytes(bytes: [u8; 8]) -> Self;
+}
+
+impl Decodable for u64 {
+    fn decode_bytes(bytes: [u8; 8]) -> Self {
+        u64::from_le_bytes(bytes)
+    }
+}
+
+impl Decodable for f64 {
+    fn decode_bytes(bytes: [u8; 8]) -> Self {
+        f64::from_le_bytes(bytes)
+    }
+}
```

Comment:
As mentioned the length depends on the datatype and for sets Relay is sending `u32`, you need to make the array length generic:

```rs
trait Decodable<const SIZE: usize> {
    const SIZE: usize = SIZE;

    fn decode_bytes(bytes: [u8; SIZE]) -> Self;
}

impl Decodable<4> for u32 {
    fn decode_bytes(bytes: [u8; Self::SIZE]) -> Self {
        Self::from_le_bytes(bytes)
    }
}

impl Decodable<8> for f64 {
    fn decode_bytes(bytes: [u8; Self::SIZE]) -> Self {
        Self::from_le_bytes(bytes)
    }
}
```

Reply:
Do rust not have something like `constexpr`?
why can't we do
```rust
trait Decodable<T> {
    const SIZE: usize = size_of<T>;

    fn decode_bytes(bytes: [u8; SIZE]) -> Self;
}
...
```